src/dataflow/src/source/postgres.rs

```diff
@@ -129,12 +133,43 @@ impl PostgresSourceReader {
     ) -> Self {
         Self {
             source_name: source_name.clone(),
+            source_tables: connector
```

Comment:
I agree this is definitely more readable, but as I originally wrote it my understanding was that the vector would be cloned and then the elements would be moved because of `into_iter()` which is why I had to `clone()` the id field to prevent a partially moved from item error. Am I correct in understanding that the only difference in efficiency then is a pointer and two uints for the Vector?

Reply:
Ok, that matches my understanding. Looking at [compiler explorer](https://rust.godbolt.org/z/sMd6c6GnP) it looks like both versions do basically the same amount of work though cloning the vector potentially makes less function calls depending on the number of items. 

Either way I agree the `HashMap::from_iter` form is clearer in multiple ways.